### PR TITLE
Add netlink support to VdpaDevice interface

### PR DESCRIPTION
**Pull request by amorenoz**
This PR is based on top of #9 . **API breaking change **
In order to add netlink support to the vdpa device API this PR does some heavy preliminary refactoring:
- kvdpa.go is renamed to device.go
- virtio-net device and vhost-vdpa device information handling is moved to their own files (virtio.go and vhost.go) and encapsulated into their own interfaces VirtioNet and VhostVdpa which can hold different information without contaminating the same API.
- Many free functions are now part of the vdpaDev struct, the end API looks like this:

```
type VdpaDevice interface {
	Driver() string
	Name() string
	VirtioNet() VirtioNet
	VhostVdpa() VhostVdpa
	ParentDevicePath() (string, error)
}
```

### CODE REVIEW

**Comment by amorenoz** [comment]
@lmilleri PTAL

File: pkg/kvdpa/device.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	*/
+	parent := filepath.Dir(devicePath)
+	if parent == rootDevDir {
+		return devicePath, nil
+	}
+
+	return parent, nil
+}
+
+/* Finds the virtio vdpa device of a vdpa device and returns it's path
```
Comment (lmilleri) [inline pkg/kvdpa/device.go]: typo, "its" instead?

**Review by lmilleri — APPROVED** [review]
/lgtm

File: pkg/kvdpa/device.go (hunk 1 above)
Comment (amorenoz) [reply pkg/kvdpa/device.go]: yep, good catch. I'll fix it

Comment (amorenoz) [reply pkg/kvdpa/device.go]: Done

**Pull request merged**